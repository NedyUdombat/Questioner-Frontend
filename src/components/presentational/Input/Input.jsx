import React from 'react';
import PropTypes from 'prop-types';

// styles
import './Input.scss';

const Input = ({
  placeholder,
  className,
  type,
  onChange,
  value,
  name,
  required,
  minLength,
  pattern,
  title,
}) => {
  return (
    <input
      placeholder={placeholder}
      className={className}
      type={type}
      onChange={onChange}
      value={value}
      name={name}
      required={required}
      minLength={minLength}
      pattern={pattern}
      title={title}
    />
  );
};

Input.propTypes = {
  className: PropTypes.string,
  placeholder: PropTypes.string,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  type: PropTypes.string.isRequired,
  value: PropTypes.string,
  pattern: PropTypes.string,
  required: PropTypes.string,
  minLength: PropTypes.string,
  title: PropTypes.string,
};

export default Input;
