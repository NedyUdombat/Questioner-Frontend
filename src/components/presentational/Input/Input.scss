@import '../../../assets/sass/index';

.form-control-input {
  border: .0625rem solid $secondary-white-2;
  border-radius: .25rem;
  color: $secondary-grey;
  font-size: 1rem;
  height: 100%;
  line-height: 1.25rem;
  padding-left: .9375rem !important;
  width: 100%;
}

.form-control-input:hover {
  border: .0625rem solid $primary-black;
}

.form-control-input:focus {
  border: .0625rem solid $primary-black;
  outline: 0;
}

.form-control-input::placeholder {
  color: $primary-grey;
  font-family: $raleway;
  font-size: 1rem;
  line-height: 1.25rem;
}
