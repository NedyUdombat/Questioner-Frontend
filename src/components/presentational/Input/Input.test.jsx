import React from 'react';
import { mount } from 'enzyme';

import Input from './Input';

describe('Input', () => {
  it('should render correctly', () => {
    const component = mount(
      <Input
        placeholder="placeholder"
        className="form-control"
        type="text"
        onChange={jest.fn()}
        value="value"
        name="name"
        required="required"
        minLength="20"
        pattern="($^+[a-z])"
        title="title"
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
