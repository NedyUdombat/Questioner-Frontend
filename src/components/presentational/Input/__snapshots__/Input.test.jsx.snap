// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<Input
  className="form-control"
  minLength="20"
  name="name"
  onChange={[MockFunction]}
  pattern="($^+[a-z])"
  placeholder="placeholder"
  required="required"
  title="title"
  type="text"
  value="value"
>
  <input
    className="form-control"
    minLength="20"
    name="name"
    onChange={[MockFunction]}
    pattern="($^+[a-z])"
    placeholder="placeholder"
    required="required"
    title="title"
    type="text"
    value="value"
  />
</Input>
`;
